pyk/src/pyk/kcfg/minimize.py

```diff
@@ -96,26 +95,19 @@ def lift_split_edge(self, b_id: NodeIdLike) -> None:
         # Ensure split can be lifted soundly (i.e., that it does not introduce fresh variables)
         assert (
             len(split_from_b.source_vars.difference(a.free_vars)) == 0
-            and len(split_from_b.target_vars.difference(split_from_b.source_vars)) == 0
+            and len(split_from_b.target_vars.difference(split_from_b.source_vars)) == 0  # <-- Can we delete this check?
```

Comment:
We might have this situation after merging node. I don't know the intent of this assertion.

Reply:
@PetarMax do you recall why this check was introduced? It seems to be introduced here: https://github.com/runtimeverification/pyk/pull/1009, where it was indicated that this "prevents a bug where split lifting could cause infinite loop", but it's difficult to tell there if a test of that particular case was added.